src/main/java/org/sonar/plugins/buildbreaker/QualityGateBreaker.java

```diff
@@ -0,0 +1,201 @@
+            return;
+        }
+
+        if (settings.getBoolean(BuildBreakerPlugin.SKIP_KEY)) {
+            LOGGER.debug("{} is disabled ({} = true)", CLASSNAME, BuildBreakerPlugin.SKIP_KEY);
+            return;
+        }
+
+        checkQualityGate();
+    }
+
+    private void checkQualityGate() {
+        Properties reportTaskProps = loadReportTaskProps();
+
+        HttpConnector httpConnector = new HttpConnector.Builder()
+                .url(reportTaskProps.getProperty("serverUrl"))
+                .credentials(settings.getString(CoreProperties.LOGIN), settings.getString(CoreProperties.PASSWORD))
+                .build();
+
+        WsClient wsClient = new HttpWsClient(httpConnector);
+
+        String ceTaskId = reportTaskProps.getProperty("ceTaskId");
+        WsRequest ceTaskRequest = new GetRequest("api/ce/task").setParam("id", ceTaskId).setMediaType(MediaTypes.PROTOBUF);
+
+        String analysisId = getAnalysisId(wsClient, ceTaskRequest);
+
+        LOGGER.debug("Requesting quality gate status for analysisId {}", analysisId);
+        ProjectStatusWsResponse projectStatusResponse = wsClient.qualityGates().projectStatus(
+                new ProjectStatusWsRequest().setAnalysisId(analysisId));
+
+        ProjectStatus projectStatus = projectStatusResponse.getProjectStatus();
+
+        Status status = projectStatus.getStatus();
+        LOGGER.info("Quality gate status: {}", status);
+
+        int errors = 0;
+        if (Status.ERROR.equals(status) || Status.WARN.equals(status)) {
+            errors = logConditions(projectStatus.getConditionsList());
+        }
+
+        if (Status.ERROR.equals(status)) {
+            LOGGER.error("{}Project did not meet {} conditions", BuildBreakerPlugin.BUILD_BREAKER_LOG_STAMP, errors);
+            fail("Project does not pass the quality gate.");
+        }
+    }
+
+    private Properties loadReportTaskProps() {
+        File reportTaskFile = new File(fileSystem.workDir(), "report-task.txt");
+
+        Properties reportTaskProps = new Properties();
+
+        try {
+            reportTaskProps.load(FileUtils.openInputStream(reportTaskFile));
+        } catch (IOException e) {
+            throw new IllegalStateException("Unable to load properties from file " + reportTaskFile, e);
+        }
+
+        return reportTaskProps;
+    }
+
+    private static String getAnalysisId(WsClient wsClient, WsRequest ceTaskRequest) {
+        WsResponse wsResponse = wsClient.wsConnector().call(ceTaskRequest);
+
+        try {
+            TaskResponse taskResponse = TaskResponse.parseFrom(wsResponse.contentStream());
+            TaskStatus taskStatus = taskResponse.getTask().getStatus();
+
+            switch (taskStatus) {
+                case IN_PROGRESS:
+                case PENDING:
+                    // Wait 10 seconds then recurse
+                    // TODO should we limit the number of attempts?
```

Comment:
I think it is safer to set a limit but I don't know what should be the value :) On very big SQ instances the CE queue may be overcrowded so the task may wait for a while before being picked. The processing itself may take some time on big projects.